gen/tests/test_adminrouter_tls_conf.py

```diff
@@ -29,25 +29,137 @@ def test_default(self):
             # `openssl` commands.
             #
             # See comments on https://jira.mesosphere.com/browse/DCOS-13437 for more
             # details.
-
             ssl_protocols TLSv1.1 TLSv1.2;
             """
         )
         assert config['content'] == expected_configuration
 
+    def test_agent(self):
+        """
+        By default, the configuration specifies certain TLS settings.
+
+        This test is a sanity check for the configuration template logic
+        rather than a particularly useful feature test.
+        """
+        config_path = '/etc/adminrouter-tls-agent.conf'
+        arguments = make_arguments(new_arguments={})
+        generated = gen.generate(arguments=arguments)
+        package = generated.templates['dcos-config.yaml']['package']
+        [config] = [item for item in package if item['path'] == config_path]
 
-class TestToggleTLS1:
+        expected_configuration = dedent(
+            """\
+            # Note that Agent Admin Router only servers cluster-internal clients. Hence,
+            # browser compatibility is not a criterion for the TLS cipher suite selection.
+            ssl_ciphers EECDH+AES256:RSA+AES256:EECDH+AES128:RSA+AES128:!MD5;
+            ssl_prefer_server_ciphers on;
+            ssl_protocols TLSv1.2;
+            """
+        )
+        assert config['content'] == expected_configuration
+
+
+class TestSetCipherOverride:
     """
-    Tests for toggling TLS 1.0.
+    Tests for setting ssl_ciphers
+
+    To test manually, either use openssl commands or sslscan
+    [https://github.com/rbsec/sslscan]
+    """
+
+    def supported_ssl_ciphers_master(self, new_config_arguments) -> List[str]:
+        """
+        Finds the line that looks like:
+        ssl_ciphers EECDH+AES256:RSA+AES256:EECDH+AES128:RSA+AES128:EECDH+3DES:RSA+3DES:!MD5;
+        and returns the list of ciphers.
+        Args:
+            new_config_arguments: Arguments which are added to the 'standard'
+                set of arguments before generating configuration files.
+        """
+        arguments = make_arguments(new_arguments=new_config_arguments)
+        generated = gen.generate(arguments=arguments)
+        package = generated.templates['dcos-config.yaml']['package']
+        config_path = '/etc/adminrouter-tls-master.conf'
+        [config] = [item for item in package if item['path'] == config_path]
+        [ssl_ciphers_line] = [
+            line for line in config['content'].split('\n') if
+            # We strip whitespace from the beginning of the line as NGINX
+            # configuration lines can start with whitespace.
+            line.lstrip().startswith('ssl_ciphers ')
+        ]
+        ssl_ciphers_line = ssl_ciphers_line.strip(';')
+        ciphers = ssl_ciphers_line.split()[1:]
+        return ciphers
 
-    To manually test that this is, in fact, a working toggle for TLS 1.0, use
+    def supported_ssl_ciphers_agent(self, new_config_arguments) -> List[str]:
```

Comment:
Both `supported_ssl_ciphers_master` and `supported_ssl_ciphers_agent` are almost identical. Can we extract single function and keep existing one calling the new function.